pkgs/os-specific/linux/kernel/manual-config.nix

```diff
@@ -312,7 +314,7 @@ stdenv.mkDerivation ((drvAttrs config stdenv.hostPlatform.linux-kernel kernelPat
       ++ optional  (lib.versionAtLeast version "4.14" && lib.versionOlder version "5.8") libelf
       # Removed util-linuxMinimal since it should not be a dependency.
       ++ optionals (lib.versionAtLeast version "4.16") [ bison flex ]
-      ++ optional  (lib.versionAtLeast version "5.2")  cpio
+      ++ optional  (lib.versionAtLeast version "5.2")  [ cpio python3 zlib.dev pahole ]
```

Comment:
You are sure about .dev in nativeBuildInputs?

Reply:
I mean, it works here. I had to specify the .dev package so the headers were available, but maybe it should go in another kind of buildInput? (I'm never 100% clear if packages should go in buildInputs, nativeBuildInputs, or elsewhere tbh).

Reply:
nativeBuildInputs is for dependencies that are supposed to be run during the build; i.e. cmake.

buildInputs is for regular dependencies whose files content is referenced by or during the build; i.e. libraries.

Whether it works or not isn't really the important bit here, it's about declaring things properly so that everything works nicely as it should.  
Cross-compilation is a good example as nativeBuildInputs need to run on the builder which has a different CPU ISA than the target. This is mostly done automatically but the system needs to be told what's what to be able to do such things.

Reply:
```suggestion
      ++ optionals  (lib.versionAtLeast version "5.2")  [ cpio python3 zlib.dev pahole ]
```

Reply:
Yeah, I really don't think `zlib.dev` is the right thing there.

You probably want to add `zlib` to `buildInputs` instead of `zlib.dev` to `nativeBuildInputs`.